controllers/storageclient_controller.go

```diff
@@ -549,6 +572,57 @@ func (r *StorageClientReconciler) reconcileSharedfileStorageClaim() error {
 	return nil
 }
 
+func (r *StorageClientReconciler) reconcileRemoteNoobaa() error {
+	// Create noobaa CR referencing the join secret
+	nb := &nbv1.NooBaa{}
+	nb.Name = "noobaa-remote"
+	nb.Namespace = r.OperatorNamespace
+
+	_, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, nb, func() error {
+		if err := r.own(nb); err != nil {
+			return err
+		}
+		utils.AddAnnotation(nb, "remote-client-noobaa", "true")
+		nb.Spec.JoinSecret = &corev1.SecretReference{
+			Name:      joinSecretName,
+			Namespace: r.OperatorNamespace,
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create remote noobaa: %v", err)
+	}
+	return nil
+}
+
+func (r *StorageClientReconciler) reconcileNoobaaJoinSecret(noobaaData map[string]string) error {
+	noobaaAuthToken, ok := noobaaData["auth_token"]
+	if !ok {
```

Comment:
What if it is found but have an empty value?

Reply:
These values will anyway be validated by remote noobaa. What I am doing here is extracting the data and creating a join secret